lib/compat/wordpress-6.2/rest-api.php

```diff
@@ -101,3 +101,18 @@ function gutenberg_register_global_styles_endpoints() {
 	$editor_settings->register_routes();
 }
 add_action( 'rest_api_init', 'gutenberg_register_global_styles_endpoints' );
+
+/**
+ * Updates REST API response for the sidebars and marks them as 'inactive'.
+ *
+ * Note: This can be a part of the `prepare_item_for_response` in `class-wp-rest-sidebars-controller.php`.
+ *
+ * @param WP_REST_Response $response The sidebar response object.
+ * @return WP_REST_Response $response Updated response object.
+ */
+function gutenberg_modify_rest_sidebars_response( $response ) {
+	$response->data['status'] = wp_is_block_theme() ? 'inactive' : 'active';
```

Comment:
I'm not sure this is working as expected. 🤔 

If the user is not using a block-based theme, the `status` is always set to active. What if the user had an inactive sidebar, isn't using the site editor yet, but updates to the new Gutenberg version? That sidebar now flips to active without the user doing anything. 

This is breaking a few core unit tests in this file: https://github.com/WordPress/wordpress-develop/blob/53063c633306997e2502a7ea90c90b58236f7183/tests/phpunit/tests/rest-api/rest-sidebars-controller.php#L276-L318

Reply:
If you know how to fix this, I'd super appreciate if we could merge something quickly so we can create a patch release for 14.9!

Reply:
Thanks for catching this, @noahtallen!

I'm AFK until Monday, but I will create a patch before release.

Reply:
Thanks, that'd be greatly appreciated!

Reply:
I opened a patch here: https://github.com/WordPress/gutenberg/pull/46966